### PR TITLE
Adding unit test integration for the boiler plate.

### PR DESCRIPTION
**Pull request by ramachandrangunasekaran**
Hi Sridhar,

I am integrating unit test integration for out boiler plate. Please let me know if any changes needed. Else, we could merge to your repository.

### CODE REVIEW

File: app/routes/user-routes.js [hunk 1]
```diff
@@ -26,7 +26,7 @@ function signUp(req, res) {
 
     emailController.sendWelcomeEmail(user.email, verificationHash);
 
-    return res.status(200).send({
+    return res.status(201).send({
```
Comment (sridharrajs) [inline app/routes/user-routes.js]: > HTTP Status Code 201: The request has been fulfilled and has resulted in one or more new resources being created. 

`201` apt for instances where the user has uploaded file. I think `200` is good for the user creation

File: .gitignore [hunk 2]
```diff
@@ -22,3 +23,8 @@ dist/
 .vscode
```
Comment (sridharrajs) [inline .gitignore]: Can you please move `.vscode` near to `.idea/` as I've grouped all the IDE meta folder together

File: test.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
+/**
+ * Created by sridharrajs.
```
Comment (sridharrajs) [inline test.js]: Please replace `sridharrajs` with `ramachandrangunasekaran` as the author of the file.

File: test.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+const MongoMockServer = require('mongodb-memory-server').MongoMemoryServer;
+const dotenv = require('dotenv').config({ path: `${__dirname}/.env.test` });
+const mInstance = new MongoMockServer();
+
+class TestFactory extends CF {
+
+  static init() {
+    if (dotenv.error) {
+      console.trace(chalk.red('.env.test file is missing'));
+      return Promise.reject('.env.test file is missing')
```
Comment (sridharrajs) [inline test.js]: missing a `;`

**Comment by ramachandrangunasekaran** [comment]
I have done the changes.

**Comment by sridharrajs** [comment]
Thanks for the PR @ramachandrangunasekaran

**Pull request merged**